ui/perfherder/CompareSubtestDistributionView.jsx

```diff
@@ -0,0 +1,304 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular/index.es2015';
+
+import dancing_cat from '../img/dancing_cat.gif';
+import perf from '../js/perf';
+import Graph from '../intermittent-failures/Graph';
+import RepositoryModel from '../models/repository';
+import PerfSeriesModel from '../models/perfSeries';
+import PushModel from '../models/push';
+
+import RevisionInformation from './RevisionInformation';
+
+// TODO remove $stateParams after switching to react router
+export default class CompareSubtestDistributionView extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dataLoading: this.props.dataLoading,
+      originalProject: null,
+      baseDrawingData: [],
+      newDrawingData: [],
+      newProject: null,
+      commonGraphSpecs: {
```

Comment:
I don't think you need this in state - you can just return it in `getGraphSpecs` and call it for each graph, adding variables in for relevant things like title, data, etc